app/models/core_data_connector/place.rb

```diff
@@ -31,14 +31,34 @@ class Place < ApplicationRecord
     # User defined fields parent
     resolve_defineable -> (place) { place.project_model }
 
-    def self.with_centroid
+    def self.centroid_function
       function = Arel::Nodes::NamedFunction.new(
         'st_centroid',
         [PlaceGeometry.arel_table[:geometry]]
       ).as('geometry_center')
+    end
+
+    def self.with_centroid
+      left_joins(:place_geometry)
+        .select(arel_table[Arel.star], with_centroid_function)
```

Comment:
Should this just be `centroid_function`?

Reply:
Yes, and it's super annoying that VS Code doesn't catch simple stuff like undefined variables, even with the Ruby LSP extension installed. It makes Ruby a lot harder to work with than JS. Does RubyMine do a better job of catching that type of thing?